### PR TITLE
Pathconf names

### LINKED ISSUE #4494: Implement os.pathconf_names
Placeholder to track the implmentation of os.pathconf_names  -- additional detail to follow
(related to Extend the os module #1175)

### PR DESCRIPTION
**Pull request by marvinmednick**
Implementation for os.pathconf_names  resolves #4494

### CODE REVIEW

**Comment by marvinmednick** [comment]
Let me know if I did something incorrectly -- I see as part my pull requesets commits authored by others....   The two changes I'm looking to pull in for pathconf_names are https://github.com/RustPython/RustPython/pull/4508/commits/d8fe7787bf9690d86bb7c57885d7fa0b0f1fb8be and https://github.com/RustPython/RustPython/pull/4508/commits/df482663717dd84b570cb7e7e0007f3b8ede021b

**Comment by fanninpm** [comment]
After CI completes, feel free to do a `git rebase ...` followed by a `git push --force`.

**Comment by youknowone** [comment]
I rebased it. `git rebase` is very useful tool to send patches.
https://docs.gitlab.com/ee/topics/git/git_rebase.html

File: extra_tests/snippets/stdlib_os.py [hunk 1]
```diff
@@ -507,3 +507,11 @@ def __exit__(self, exc_type, exc_val, exc_tb):
 
     for arg in [None, 1, 1.0, TabError]:
         assert_raises(TypeError, os.system, arg)
+
+# Testing for os.pathconf_names
+if "win" not in sys.platform:
```
Comment (youknowone) [inline extra_tests/snippets/stdlib_os.py:512]:
```suggestion
if not sys.platform.startswith("win"):
```
"darwin" also contains "win", which means macOS.

File: vm/src/stdlib/posix.rs [hunk 2]
```diff
@@ -1881,6 +1883,22 @@ pub mod module {
         pathconf(PathOrFd::Fd(fd), name, vm)
     }
 
+    #[pyattr]
+    fn pathconf_names(vm: &VirtualMachine) -> PyDictRef {
+        let pathname = vm.ctx.new_dict();
+        for variant in PathconfVar::VARIANTS {
+            // get the name of variant as a string to use as the dictionary key
+            let key: PyObjectRef = vm.ctx.new_str(variant.to_string()).into();
```
Comment (youknowone) [inline vm/src/stdlib/posix.rs:1891]:
```suggestion
            let key = vm.ctx.new_str(variant.to_string());
```
`PyStrRef` itself is dict key compatible.

**Review by youknowone — CHANGES_REQUESTED** [review]

**Comment by marvinmednick** [comment]
> I rebased it. `git rebase` is very useful tool to send patches. https://docs.gitlab.com/ee/topics/git/git_rebase.html

Thanks -- I was looking at trying to do that but it didn't seem to look right as I still ended up with 2 additional changes -- I was thinking my baseline wasn't quite right or I did something incorrectly.

**Comment by marvinmednick** [comment]
I'm looking at the error --    I found an online free online macos simlulator and was able to get the error 22 manually trying the command os.pathconf('/',21)   but that isn't conclusive.   (pathconf worked for indexes 0-20).   
I'd like to check the the PathconfVar entries for darwin...   but darwin doesn't seem to be a confguration check within PathconfVar    

It looks like there are some platforms which may have pathconf indexes that are greater then 20  (e.g.  in libc  
libc:   src/unix/haiku/mod.rs:pub const _PC_REC_MAX_XFER_SIZE: ::c_int = 32;
though on linux the same parameter is index 15.
libc    src/unix/linux_like/linux/mod.rs:pub const _PC_REC_MAX_XFER_SIZE: ::c_int = 15;

I suppose this is more a of libc specfic question does anyone know how can I tell which of the libc flavor will be used on macos/darwin? (and therefore which pathconf indexes value will show up in PathconfVar)

File: vm/src/stdlib/posix.rs [hunk 3]
```diff
@@ -1881,6 +1883,24 @@ pub mod module {
         pathconf(PathOrFd::Fd(fd), name, vm)
     }
 
+    // TODO: this is expected to be run on macOS as a unix, but somehow not.
```
Comment (youknowone) [inline vm/src/stdlib/posix.rs:1886]: I put linux cfg until it is fixed for macos

**Review by youknowone — APPROVED** [review]
You don't need to fix every macOS issue if if you don't have macOS machine. Please fill free to ask help if you have hard trouble to fix.

I think adding it only for linux is a working option. I made a change. We can fix macOS issue later.

Thank you for contributing!

**Comment by youknowone** [comment]
Thank you for contributing!

**Pull request merged**